doc/source/dev/conduct/code_of_conduct.rst

```diff
@@ -0,0 +1,164 @@
+   within the context of our shared goal of improving SciPy project code,
+   documentation and other content.
+
+4. Be careful in the words that we choose. Whether we are participating as
+   professionals or volunteers, we value professionalism in all interactions,
+   and take responsibility for our own speech. Be kind to others. Do not insult
+   or put down other participants. Harassment and other exclusionary behaviour
+   are not acceptable. This includes, but is not limited to:
+
+    - Violent threats or language directed against another person.
+    - Sexist, racist, or otherwise discriminatory jokes and language.
+    - Posting sexually explicit or violent material.
+    - Posting (or threatening to post) other people's personally identifying information ("doxing").
+    - Sharing private content, such as emails sent privately or non-publicly, or unlogged forums such as IRC channel history.
+    - Personal insults, especially those using racist or sexist terms.
+    - Unwelcome sexual attention.
+    - Excessive or unnecessary profanity.
+    - Repeated harassment of others. In general, if someone asks you to stop, then stop.
+    - Advocating for, or encouraging, any of the above behaviour.
+
+
+Diversity Statement
+-------------------
+
+The SciPy project welcomes and encourages participation by everyone. We are
+committed to being a community that everyone feels good about joining. Although
+we may not be able to satisfy everyone, we will always work to treat everyone
+well.
+
+No matter how you identify yourself or how others perceive you: we welcome you.
+Though no list can hope to be comprehensive, we explicitly honour diversity in:
+age, culture, ethnicity, genotype, gender identity or expression, language,
+national origin, neurotype, phenotype, political beliefs, profession, race,
+religion, sexual orientation, socioeconomic status, subculture and technical
+ability.
+
+Though we welcome people fluent in all languages, SciPy development is
+conducted in English.
+
+Standards for behaviour in the SciPy community are detailed in the Code of
+Conduct above. We expect participants in our community to meet these standards
+in all their interactions and to help others to do so as well.
+
+
+Reporting Guidelines
+--------------------
+
+While this code of conduct should be adhered to by participants, we recognize
+that sometimes people may have a bad day, or be unaware of some of the
+guidelines in this code of conduct. When that happens, you may reply to them
+and point out this code of conduct. Such messages may be in public or in
+private, whatever is most appropriate. However, regardless of whether the
+message is public or not, it should still adhere to the relevant parts of this
+code of conduct; in particular, it should not be abusive or disrespectful.
+Assume good faith; it is more likely that participants are unaware of their bad
+behaviour than that they intentionally try to degrade the quality of the
+discussion.
```

Comment:
As currently phrased, these reporting guidelines give the impression that the onus is on whoever received the violation to confront the offender, and that it hasn't occurred to us that people could just be assholes. These are both really problematic IMO. This is fine for cases where someone "merely" goes a bit over the line and the person observing it feels comfortable confronting them about it, but that definitely doesn't cover all cases.

One obvious case is when someone is simply being rude and abusive and definitely not acting in good faith. Last night someone logged into the `#scipy` channel on Freenode and started pasting racial slurs in giant letters. The text above makes it seem like we will scold people for not attempting to engage with such people in "good faith". (I certainly didn't try engaging with them; I joined the `#freenode` channel and checked that they'd been banned from the network.) Hopefully this never happens in our community, but in other communities it has happened that people randomly get private emails containing threats and abuse from people that behave reasonably in public; recipients may not feel safe confronting such people themselves. (Common advice in such cases is to not reply in case of further aggravating the situation.) It's sadly common for women in particular to receive sexist harrassment and then be told that it's their fault and that the institutions that are supposed to support them (like a CoC committee) are not going to do so because they didn't respond in the "right way" -- the current phrasing of this section may give the impression that their options are either to risk their safety trying to handle the situation themselves, or else we won't have their back. It's also very common for institutions to downplay this kind of harassment ("oh it can't have been that bad", "he's a good guy really (when he's not raping someone)", etc.); if we give the impression that we believe the only reason anyone might violate the CoC is that they're having a "bad day", then we may also give the impression that we can't be trusted to handle more serious situations appropriately.

> it is more likely that participants are unaware of their bad behaviour than that they intentionally try to degrade the quality of the discussion.

I don't even believe this is true. I've certainly encountered more cases like the racial-slur-shouting-IRC-person than I have cases where someone acting in good faith nonetheless managed to violate social norms to such an extent that I'd think about invoking a CoC, and folks with e.g. female-coded names receive *way* more explicit abuse than I do.

We should follow other CoCs like [Django's](https://www.djangoproject.com/conduct/reporting/) or [WeAllJS](https://wealljs.org/code-of-conduct) and make clear that while people certainly can confront offenders themselves, there's absolutely no obligation to do so. We should also avoid giving the impression that of-course-we're-all-friends-here; it's just not true and it undermines the whole document.

Honestly we could do worse than to just copy Django's reporting guide.

Reply:
I agree that this paragraph does not adequately cover the options; while honest mistakes do occur, it is best for the person on the receiving end to decide whether that was the case, and to report behavior they consider abusive at their prerogative.

I'm not sure how I feel about Django's encouragement to over-report.  I guess that's fine if you have a committee you trust to truly "forget" incidents deemed non-violations?  I lean towards making it easy to report, seriously encouraging reporting when there's a feeling of having been treated poorly, but also trying to work thing out if that seems like a manageable & low-risk route.

I don't think the IRC example is a great one, because that behavior is so far to one side of the spectrum.  The sticky situations arise when there's a lot more uncertainty around how to interpret actions and intent.  The response should be commensurate with the transgression.

Reply:
We discussed this extensively on the call today. 

> if we give the impression that we believe the only reason anyone might violate the CoC is that they're having a "bad day"

Clearly we don't believe that, so let's ensure the text doesn't give that impression.

Reply:
> The response should be commensurate with the transgression.

+1